src/WebJobs.Script/Host/WorkerFunctionMetadataProvider.cs

```diff
@@ -7,90 +7,117 @@
+                {
+                    if (channels.TryGetValue(workerId, out TaskCompletionSource<IRpcWorkerChannel> languageWorkerChannelTask))
                     {
-                        functions = ValidateMetadata(rawFunctions);
-                    }
+                        _logger.LogDebug("Found initialized language worker channel for runtime: {workerRuntime} workerId:{workerId}", _workerRuntime, workerId);
+                        try
+                        {
+                            IRpcWorkerChannel channel = await languageWorkerChannelTask.Task;
+                            rawFunctions = await channel.GetFunctionMetadata();
+
+                            if (IsDefaultIndexingRequired(rawFunctions))
+                            {
+                                _functions.Clear();
+                                return new FunctionMetadataResult(useDefaultMetadataIndexing: true, _functions);
+                            }
 
-                    // set up invocation buffers and send load requests
-                    await _dispatcher.FinishInitialization(functions);
+                            if (!IsNullOrEmpty(rawFunctions))
+                            {
+                                functions = ValidateMetadata(rawFunctions);
+                            }
 
-                    // Validate if the app has functions in legacy format and add in logs to inform about the mixed app
-                    _ = Task.Delay(TimeSpan.FromMinutes(1)).ContinueWith(t => ValidateFunctionAppFormat(_scriptOptions.Value.RootScriptPath, _logger, environment));
+                            _functions = functions.ToImmutableArray();
+                            _logger.FunctionMetadataProviderFunctionFound(_functions.IsDefault ? 0 : _functions.Count());
+
+                            // Validate if the app has functions in legacy format and add in logs to inform about the mixed app
+                            _ = Task.Delay(TimeSpan.FromMinutes(1)).ContinueWith(t => ValidateFunctionAppFormat(_scriptOptions.Value.RootScriptPath, _logger, environment));
+
+                            break;
+                        }
+                        catch (Exception ex)
+                        {
+                            _logger.LogWarning(ex, "Removing errored webhost language worker channel for runtime: {workerRuntime} workerId:{workerId}", _workerRuntime, workerId);
+                            await _channelManager.ShutdownChannelIfExistsAsync(_workerRuntime, workerId, ex);
+                        }
+                    }
                 }
-                else
+            }
+
+            return new FunctionMetadataResult(useDefaultMetadataIndexing: false, _functions);
+        }
+
+        private bool IsDefaultIndexingRequired(IEnumerable<RawFunctionMetadata> functions)
+        {
+            foreach (RawFunctionMetadata function in functions)
+            {
+                if (function.UseDefaultMetadataIndexing == true)
```

Comment:
Could simplify to 
```suggestion
                if (function.UseDefaultMetadataIndexing)
```
or 
```suggestion
                return functions.Any(f => f.UseDefaultMetadataIndexing);
```
?